src/store/modules/nutrition/actions.ts

```diff
@@ -11,17 +14,53 @@ export const actions: ActionTree<any, any> = {
                     day: payload,
                 });
             } else {
-                new NutritionAPI()
-                    .getMock(payload)
-                        .then((v: IDailyIntake) => {
-                            commit('cacheDailyIntake', {
-                                day: payload,
-                                intake: v,
-                            });
-                            commit('getDailyIntake', {
-                                day: payload,
-                            });
-                        });
+                try {
+                    const intake: IDailyIntake = await new NutritionAPI().getMock(payload) || {} as IDailyIntake;
+                    commit('cacheDailyIntake', {
+                        day: payload,
+                        intake,
+                    });
+                    commit('getDailyIntake', {
+                        day: payload,
+                    });
+                    return intake;
+                } catch (error) {
+                    throw (error);
+                }
             }
     },
+    async updateCurrentDailyIntake(
+        {state, commit}:
+            {state: NutritionState, commit: Commit},
+        payload:
+            {mealName: string, food: Food}) {
+                try {
+                    const updatedDailyIntake: IDailyIntake =
+                        await new NutritionAPI().
+                            updateCurrentDailyIntake(state.currentIntake, payload.mealName, payload.food);
+                    commit('cacheDailyIntake', {
+                        day: updatedDailyIntake.day,
+                        intake: updatedDailyIntake,
+                    });
+                    commit('getDailyIntake', {
+                        day: updatedDailyIntake.day,
+                    });
+                    return updatedDailyIntake;
+                } catch (error) {
+                    throw (error);
+                }
+    },
+    async getFoodInfo({commit, getters}: {commit: Commit, getters: any}, payload: string): Promise<FoodInfo> {
+        const cachedFoodInfo: FoodInfo = getters.foodInfo(payload);
+        if (cachedFoodInfo) { return Promise.resolve(cachedFoodInfo); }
```

Comment:
I would add some line breakes here for better readability :)

Reply:
Done